### PR TITLE
modified_db_cluster based on engine type

### LINKED ISSUE #8769: modify_db_cluster method for Rds-cluster is not working when RDS EngineMode'= 'serverless'
### Describe the bug

when there are serverless v1 (EngineMode'= 'serverless') instance for RDS-cluster and we try to modify **BackupRetentionPeriod**.It throws an error saying 

botocore.exceptions.ClientError: An error occurred (InvalidParameterCombination) when calling the ModifyDBCluster operation: You currently can't modify BackupWindow with Aurora Serverless.
--------------------------------------------------------------------------------------------------------------------------- 

### What did you expect to happen?

It should modify the "BackupRetentionPeriod".

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

_No response_

### Policy

_No response_

### Relevant log/traceback output

```shell
ERROR    custodian.output:output.py:336 Error while executing policy
Traceback (most recent call last):
  File "/Users/lfh512/CLOUDCUSTODIAN/cloud-custodian/c7n/policy.py", line 364, in run
    results = a.process(resources)
  File "/Users/lfh512/CLOUDCUSTODIAN/cloud-custodian/c7n/resources/rdscluster.py", line 226, in process
    self.process_snapshot_retention(client, cluster)
  File "/Users/lfh512/CLOUDCUSTODIAN/cloud-custodian/c7n/resources/rdscluster.py", line 234, in process_snapshot_retention
    self.set_retention_window(
  File "/Users/lfh512/CLOUDCUSTODIAN/cloud-custodian/c7n/resources/rdscluster.py", line 245, in set_retention_window
    _run_cluster_method(
  File "/Users/lfh512/CLOUDCUSTODIAN/cloud-custodian/c7n/resources/rdscluster.py", line 295, in _run_cluster_method
    method(**params)
  File "/Users/lfh512/custodian/0.9.13.0/rule-engine/.tox/py39/lib/python3.9/site-packages/botocore/client.py", line 386, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/lfh512/custodian/0.9.13.0/rule-engine/.tox/py39/lib/python3.9/site-packages/botocore/client.py", line 705, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidParameterCombination) when calling the ModifyDBCluster operation: You currently can't modify BackupWindow with Aurora Serverless.
DEBUG    placebo.pill:pill.py:175 stopping, mode=record
```

### Extra information or context

This is happening because of this limitation 

Currently, when you are updating the stack for an Aurora Serverless DB cluster, you can't include changes to any other properties when you specify one of the following properties: PreferredBackupWindow, PreferredMaintenanceWindow, and Port. This limitation doesn't apply to provisioned DB clusters.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html.

We would need a to modify_db_cluster method based on types of EngineMode, If the engine mode is "Provisioned" the existing method would work but for **EngineMethod="serverless**" we need to remove **PreferredBackupWindow** and **PreferredMaintenanceWindow**

### PR DESCRIPTION
**Pull request by shineyb**
closes #8769

### CODE REVIEW

**Comment by shineyb** [comment]
This PR closes this [issue](https://github.com/cloud-custodian/cloud-custodian/issues/8769)

File: c7n/resources/rdscluster.py [hunk 1]
```diff
@@ -244,15 +243,23 @@ def process_snapshot_retention(self, client, cluster):
...
+                dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
+                    BackupRetentionPeriod=retention),
+                (client.exceptions.DBClusterNotFoundFault, client.exceptions.ResourceNotFoundFault),
+                client.exceptions.InvalidDBClusterStateFault)
+        else:
+            _run_cluster_method(
+                client.modify_db_cluster,
+                dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
+                    BackupRetentionPeriod=retention,
+                    PreferredBackupWindow=cluster['PreferredBackupWindow'],
```
Comment (PratMis) [inline c7n/resources/rdscluster.py:259]: This action is for setting log retention. I was wondering if we even needed reapplying `PreferredBackupWindow` and `PreferredMaintenanceWindow` at all re: modifying log retention. I tested it with serverless and aurora-mysql, modifying `BackupRetentionPeriod` works without reapplying the other two configuration. However, I might have to do more investigation wrt other engine types

Comment (shineyb) [reply c7n/resources/rdscluster.py:259]: So this is specifically for Serverless v1 which is kind of deprecated. However you can still create it using this [link](https://catalog.us-east-1.prod.workshops.aws/workshops/098605dc-8eee-4e84-85e9-c5c6c9e43de2/en-US/lab10-aurora-serverless/setup).

[Here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html) is the limitation on v1. 

_Currently, when you are updating the stack for an Aurora Serverless DB cluster, you can't include changes to any other properties when you specify one of the following properties: PreferredBackupWindow, PreferredMaintenanceWindow, and Port. This limitation doesn't apply to provisioned DB clusters._

Comment (PratMis) [reply c7n/resources/rdscluster.py:259]: yup, i got that. My comment was more from an implementation pov. Currently, we have:
```
        _run_cluster_method(
            client.modify_db_cluster,
            dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
                 BackupRetentionPeriod=retention,
                 PreferredBackupWindow=cluster['PreferredBackupWindow'],
                 PreferredMaintenanceWindow=cluster['PreferredMaintenanceWindow']),
```
I was thinking whether the 2 parameters aka `PreferredBackupWindow` and `PreferredMaintenanceWindow` were required as part of this action?
Could we just do?
```
        _run_cluster_method(
            client.modify_db_cluster,
            dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
                 BackupRetentionPeriod=retention),
```

**Pull request merged**